operator/api/redpanda/v1alpha2/redpanda_types.go

```diff
@@ -309,5 +297,5 @@ func (in *Redpanda) GetDot(restConfig *rest.Config) (*helmette.Dot, error) {
 		IsUpgrade: true,
 	}
 
-	return redpandachart.Chart.Dot(restConfig, release, partial)
+	return redpandachart.Chart.Dot(restConfig, release, in.Spec.ClusterSpec)
```

Comment:
It's possible that this comes back to bite us, especially with the now closed chart schemas.

Could we get away with something like:
```go
spec := in.Spec.ClusterSpec.DeepCopy()

if spec.DeprecatedFullNameOverride != nil && spec.FullnameOverride == nil {
spec.FullnameOverride = spec.DeprecatedFullNameOverride
}

// write a comment about why this is done.
spec.DeprecatedFullNameOverride = nil

// Round trip through values to prune any unknown/unused values.
// ...
```

Reply:
It will not work. The rest of the Redpanda chart written in Go will ignore `spec.DeprecatedFullNameOverride` as the round trip is not implemented in Redpanda reconciler.

https://github.com/redpanda-data/redpanda-operator/blob/6da70996b25ddc8f39e0582ab593dabbd0fc79fb/operator/internal/controller/redpanda/redpanda_controller.go#L290-L299

Under the hood Render will take `any` as a parameter and yaml library will ignore `DeprecatedFullNameOverride` as it does not match the json struct annotation. The function I'm changing is used in other part of the reconciliation where this json round trip is done. The name of the certificate is changed, so the reconciliation is failing.